src/App.js

```diff
@@ -54,17 +39,17 @@ function App({ data }) {
               <Redirect to="/" />
             ) : (
               <BookList
-                books={filteredBooks}
-                readingStatus={readingStatus}
+                filteredBooks={filteredBooks}
```

Comment:
not related to the pull request: instead of checking the username in every route you can do it outside, either in `<AppContainer>` or, since it's a redirect add a condionial `return(<Redirect ... >)` before the real return

Reply:
Thank you! I tried your suggested options and the first one didn't work directly, because of the redirect. I tried also the second option and the result was:
```js
if (!username) {
    if (pathname === '/') {
      return (
        <AppContainer>
          <Start setUsername={setUsername} />
        </AppContainer>
      )
    } else {
      return <Redirect to="/" />
    }
  } else {
    return (
      <AppContainer>
        <Switch>
          <Route exact path="/">
            <Redirect to="/currently-reading" />
          </Route>
          <Main>
            <Route exact path={['/currently-reading', '/library']}>
              <BookList
                filteredBooks={filteredBooks}
                username={username}
                status={pathname}
              />
            </Route>
            <Route exact path="/add-book">
              <AddBook books={books} setBooks={setBooks} username={username} />
            </Route>
          </Main>
        </Switch>
        <Route exact path={['/currently-reading', '/library', '/add-book']}>
          <Footer>
            <Navigation />
          </Footer>
        </Route>
      </AppContainer>
    )
  }
``
For me this code is less readable than the solution which had the username check for every route even so that one is more repetitive. 🤔  The solution above also won't reduce the code itself. That's why I would prefer the solution with checking the username in every route if it's okay for you. :-)